pkg/registry/secret/strategy.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2015 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package secret
+
+import (
+	"fmt"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api/validation"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/fields"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/registry/generic"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util/fielderrors"
+)
+
+// strategy implements behavior for Secret objects
+type strategy struct {
+	runtime.ObjectTyper
+	api.NameGenerator
+}
+
```

Comment:
That's fair, though those assertions are really more for humans reading the
code than the compiler.

On Tue, Apr 28, 2015 at 12:42 PM, Jordan Liggitt <notifications@github.com>
wrote:

> In pkg/registry/secret/strategy.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/7419#discussion_r29262127>
> :
>
> > +
> > +	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
> > +	"github.com/GoogleCloudPlatform/kubernetes/pkg/api/validation"
> > +	"github.com/GoogleCloudPlatform/kubernetes/pkg/fields"
> > +	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
> > +	"github.com/GoogleCloudPlatform/kubernetes/pkg/registry/generic"
> > +	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
> > +	"github.com/GoogleCloudPlatform/kubernetes/pkg/util/fielderrors"
> > +)
> > +
> > +// strategy implements behavior for Secret objects
> > +type strategy struct {
> > +	runtime.ObjectTyper
> > +	api.NameGenerator
> > +}
> > +
>
> compiler would catch it when assigned at
> https://github.com/GoogleCloudPlatform/kubernetes/pull/7419/files#diff-b9cf9c3ecb5fce05f0eb0ee398d2ed0fR61
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/7419/files#r29262127>
> .
>